app/controllers/insured/group_selection_controller.rb

```diff
@@ -246,6 +245,11 @@ def calculate_elected_aptc_pct(aptc_applied_amount, aggregate_aptc_amount)
     (aptc_applied_amount / aggregate_aptc_amount).round(2)
   end
 
+  def is_user_authorized?
+    primary_family_enrollments = current_user.person.primary_family.hbx_enrollments.map(&:id).map(&:to_s)
```

Comment:
why don't search the hbx_enrollment via the database?

Reply:
we do that further in the process. We were completely safe if the `hbx_enrollment` does not exist in the database but if it exists then we are trying to find if the HBX ID belongs to this user.